From 017dfcb144c557bb4bc72669168ad7b70a34ce57 Mon Sep 17 00:00:00 2001
From: Seo Suchan <tjtncks@gmail.com>
Date: Fri, 3 May 2024 17:18:46 +0900
Subject: [PATCH 1/2] mbedtls: fix compile on mbedtls 3.6+ they moved
 mbedtls_x509_get_name into interal zone, this declares it again in a file I
 looked approperate to hold one

Signed-off-by: Seo Suchan <tjtncks@gmail.com>
---
 lib/tls/mbedtls/private-lib-tls-mbedtls.h | 5 +++++
 1 file changed, 5 insertions(+)

--- a/lib/tls/mbedtls/private-lib-tls-mbedtls.h
+++ b/lib/tls/mbedtls/private-lib-tls-mbedtls.h
@@ -57,3 +57,8 @@ lws_tls_mbedtls_cert_info(mbedtls_x509_c
 int
 lws_x509_get_crt_ext(mbedtls_x509_crt *crt, mbedtls_x509_buf *skid,
 		     lws_mbedtls_x509_authority *akid);
+
+/* redefine hidden mbedtls internal functions*/
+int
+mbedtls_x509_get_name(unsigned char **p, const unsigned char *end,
+                          mbedtls_x509_name *cur);
